civis/parallel.py

```diff
@@ -730,7 +730,11 @@ def _fetch_result(fut):
                 fut.remote_func_output = exc
             else:
                 fut.result_fetched = True
-                if not fut.cancelled() and not fut.exception():
+                cancelled = (fut.cancelled()
+                             or (hasattr(fut, 'is_cancel_requested')
```

Comment:
Looking at the `is_cancel_requested` attribute here seems reasonable to me since platform doesn't report a job as cancelled until well after the cancellation request is made.

I think it'd be more typical to do the following here, though, rather than using `hasattr`

```
cancelled = fut.cancelled()
try:
    cancelled |= fut.is_cancel_requested
except AttributeError:
    pass
```

That said, I think it'd be best not to create a new attribute `is_cancel_requested` on the `Future` object. Could we check the run info here to get that from the API?